parquet-protobuf/src/main/java/org/apache/parquet/proto/ProtoSchemaConverter.java

```diff
@@ -79,12 +80,20 @@ public MessageType convert(Class<? extends Message> protobufClass) {
   }
 
   /* Iterates over list of fields. **/
-  private <T> GroupBuilder<T> convertFields(GroupBuilder<T> groupBuilder, List<FieldDescriptor> fieldDescriptors) {
+  private <T> GroupBuilder<T> convertFields(GroupBuilder<T> groupBuilder, List<FieldDescriptor> fieldDescriptors, List<String> parentNames) {
     for (FieldDescriptor fieldDescriptor : fieldDescriptors) {
-      groupBuilder =
-          addField(fieldDescriptor, groupBuilder)
+      final String name = fieldDescriptor.getFullName();
+      final List<String> newParentNames = new ArrayList<>(parentNames);
+      newParentNames.add(name);
+      if (parentNames.contains(name)) {
+        // Circular dependency, skip
+        LOG.warn("Breaking circular dependency:{}{}", System.lineSeparator(),
```

Comment:
I am not very familiar with Proto. By design, is the 'circular' normal in the proto or it is caused by issues?

Reply:
It is possible to create circular dependencies, that is the problem. I am not sure in what case they would be useful, however since they can exist, parquet should not fail with `StackOverflowError` when it encounters them 😄

Reply:
In that case, we silently break the circle without throwing an exception. Is  that OK?

Reply:
Well, another option would be to add a new configuration setting to allow the user to either have it fail with a good error message, or just silently break the circle like this. However I am not familiar with how `parquet-protobuf` is configured. If I should go that route, I'd appreciate some examples!

Reply:
i had been working on this issue as well and arrived at a similar solution to this one (however, without skipping/losing data) and linked to the prs in this pr conversation.  ptal, and if you folks prefer it, i can submit a merge against head and close out this pr.